File: chapter3/011.go
package main

import (
	"fmt"
)

func main() {
	var n int
	fmt.Scan(&n)

	var results []int
	for i := 2; i <= n; i++ {
		flg := true
		for j := 2; j <= i-1; j++ {
			if i%j == 0 {
				flg = false
				break
			}
		}
		if flg {
			results = append(results, i)
		}
	}

	for i, result := range results {
		fmt.Print(result)
		if i != len(results) {
			fmt.Print(" ")
		} else {
			fmt.Println("")
		}
	}
}

File: chapter3/011.rb
n = STDIN.gets.chomp.to_i

results = []
(2..n).each do |i|
  flg = true
  (2..(i - 1)).each do |ii|
    if i % ii == 0
      flg = false
      break
    end
  end
  results << i if flg
end

puts results.join(" ")
